synapse/handlers/sync.py

```diff
@@ -2353,6 +2333,34 @@ class SyncResultBuilder:
     to_device: List[JsonDict] = attr.Factory(list)
 
 
+def _calculate_user_changes(
+    sync_result_builder: "SyncResultBuilder",
+) -> Tuple[Set[str], Set[str]]:
```

Comment:
I wonder if this should be start of `SyncResultBuilder`?

Reply:
Not sure what you mean here, sorry. Perhaps that this ought to be placed above the definition of `SyncResultBuilder`?

Oh, actually I think you mean the quotes are unnecessary.

(My personal preference would be to move SyncResultBuilder and all other POD structs/classes all the way up to the top of the file, so the quotes aren't needed throughout.)

Reply:
Trimmed quotes in dd39d8276.

Reply:
What I meant to say was:

> I wonder if this should be *part* of `SyncResultBuilder`?

As in a method on that class! Clearly I failed to say anything reasonable.

Reply:
Ahh, I see. I considered this, but opted to make `_calculate_user_changes` a free function, to try and suggest that it's just reading data and not mutating the `SyncResultBuilder` instance.

I think this is just an old C++ habit dying hard though. Happy to go with whatever you think is clearest!

Reply:
I think it is fine either way, `SyncResultBuilder` seems to just be a way to pass around mutable state so probably best to keep logic out of it.